### PR TITLE
JSON-LD in responses

### LINKED ISSUE #54: Add JSON-LD support on HTTP requests-responses
The [documentation](https://bbp-nexus.epfl.ch/dev/docs/kg/api-reference/organizations.html) correctly defines the HTTP format that has to be used in the HTTP requests and returned in the responses.

However, in many the cases we are not using  the context right now, neither having proper JSON-LD predicates.

### PR DESCRIPTION
**Pull request by hygt**
Fixes #54

### CODE REVIEW

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/io/RoutesEncoder.scala [hunk 1]
```diff
@@ -26,52 +27,110 @@ abstract class RoutesEncoder[Id, Reference, Entity](base: Uri)(implicit le: Enco
...
       )
     }
 
   implicit def scoredQueryResultEncoder(implicit E: Encoder[Id]): Encoder[ScoredQueryResult[Id]] =
     Encoder.encodeJson.contramap { qr =>
       Json.obj(
-        "resultId" -> Json.fromString(qr.source.qualifyAsString),
-        "score"    -> Json.fromFloatOrString(qr.score),
-        "source"   -> E(qr.source)
+        nxvResultId -> Json.fromString(qr.source.qualifyAsString),
```
Comment (umbreak) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/io/RoutesEncoder.scala]: nxv prefix not needed here. Same as above

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/io/RoutesEncoder.scala [hunk 2]
```diff
@@ -26,52 +27,110 @@ abstract class RoutesEncoder[Id, Reference, Entity](base: Uri)(implicit le: Enco
...
+        nxvScore    -> Json.fromFloatOrString(qr.score),
+        nxvSource   -> E(qr.source)
```
Comment (umbreak) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/io/RoutesEncoder.scala]: nxv prefix not needed here. Same as above

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/io/RoutesEncoder.scala [hunk 3]
```diff
@@ -26,52 +27,110 @@ abstract class RoutesEncoder[Id, Reference, Entity](base: Uri)(implicit le: Enco
...
+    val `@context`     = "@context"
+    val `@id`          = "@id"
+    val links          = "links"
+    val nxvNs          = "nxv"
+    val nxvRev         = "nxv:rev"
+    val nxvValue       = "nxv:value"
+    val nxvDeprecated  = "nxv:deprecated"
+    val nxvDescription = "nxv:description"
+    val nxvPublished   = "nxv:published"
+    val nxvResultId    = "nxv:resultId"
```
Comment (umbreak) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/io/RoutesEncoder.scala]: nxv prefix not needed here, neither in `nxvSource ` or `nxvScore `

File: modules/service/src/main/resources/app.conf [hunk 4]
```diff
@@ -90,6 +90,9 @@ app {
 
   # Collection of prefixes used throughout the service
   prefixes {
+    # Nexus core context definition
+    core-context = ${app.http.public-uri}"/"${app.http.prefix}"/contexts/nexus/core/resource/v1.0.0"
```
Comment (umbreak) [inline modules/service/src/main/resources/app.conf]: We probably don't need that, but just the link to `standards` context, `links` context and `search` context

File: modules/core/src/test/resources/optional-int-value-schema.json [hunk 5]
```diff
@@ -52,6 +52,7 @@
       "@reverse": "http://www.w3.org/2000/01/rdf-schema#isDefinedBy",
       "@type": "@id",
       "@container": "@set"
-    }
+    },
+    "nxv": "http://localhost/v0/contexts/nexus/core/resource/v1.0.0"
```
Comment (umbreak) [inline modules/core/src/test/resources/optional-int-value-schema.json]: standards instead "{{base}}/contexts/nexus/core/standards/v0.1.0" ?

File: modules/service/src/test/scala/ch/epfl/bluebrain/nexus/kg/service/io/JsonOpsSpec.scala [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        "nxv:rev" -> Json.fromLong(1)
+      ),
+    Json.obj(
+      "@context" -> Json.obj("foo" -> Json.fromString("http://foo.domain/some/context"),
+                             "bar" -> Json.fromString("http://bar.domain/another/context")),
+      "@id"     -> Json.fromString("foo-id"),
+      "nxv:rev" -> Json.fromLong(1)
+    ) ->
+      Json.obj(
+        "@context" -> Json.obj("foo" -> Json.fromString("http://foo.domain/some/context"),
```
Comment (umbreak) [inline modules/service/src/test/scala/ch/epfl/bluebrain/nexus/kg/service/io/JsonOpsSpec.scala]: A `@context` should be converted always to an array, except for when we didn't have previously any context. And what should be added is the link to "{{base}}/contexts/nexus/core/standards/v0.1.0".
So, input like
```json
"@context" {
"a": "http://localhost/a",
"b": "http://localhost/b"
}
```
to 
```json
"@context" [
{
  "a": "http://localhost/a",
  "b": "http://localhost/b"
},
"{{base}}/contexts/nexus/core/standards/v0.1.0"
]
```

**Comment by hygt** [comment]
Jenkins retest this please

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/io/RoutesEncoder.scala [hunk 7]
```diff
@@ -26,52 +27,110 @@ abstract class RoutesEncoder[Id, Reference, Entity](base: Uri)(implicit extractI
...
+  object JsonLDKeys {
+    val `@context`     = "@context"
+    val `@id`          = "@id"
+    val links          = "links"
+    val resultId       = "resultId"
+    val source         = "source"
+    val score          = "score"
+    val nxvNs          = "nxv"
+    val nxvRev         = "nxv:rev"
+    val nxvValue       = "nxv:value"
```
Comment (umbreak) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/io/RoutesEncoder.scala]: Where is that used?

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/BootstrapService.scala [hunk 8]
```diff
@@ -188,6 +189,8 @@ object BootstrapService {
         "published",
         "deprecated",
         "links",
+        "rel",
```
Comment (umbreak) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/BootstrapService.scala]: `rel` and `href` are not used anymore when changing to an object based `links`

Comment (umbreak) [reply modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/BootstrapService.scala]: we should add `nxv:rev`, `nxv:deprecated` and `nxv:published` and the ones related to search

Comment (umbreak) [reply modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/BootstrapService.scala]:
```
 List(
        "@context",
        "@id",
        "@type",
        "total",
        "",
        "nxv:rev",
        "nxv:published",
        "nxv:deprecated",
        "links",
      )
```

**Pull request merged**